pkg/controller/pod/pod_controller.go

```diff
@@ -221,53 +220,40 @@ func (r *ReconcilePod) Reconcile(request reconcile.Request) (reconcile.Result, e
 }
 
 func (r *ReconcilePod) recreateNsxNcpResourceIfDeleted(resName string) error {
-	instance := identifyAndGetInstance(resName)
-	instanceDetails := types.NamespacedName{
-		Namespace: operatortypes.NsxNamespace,
-		Name:      resName,
-	}
-
-	doesResExist, err := r.checkIfK8sResourceExists(instance, instanceDetails)
+	doesResExist, err := operatortypes.CheckIfNCPK8sResourceExists(
+		r.client, resName)
 	if err != nil {
 		log.Error(err, fmt.Sprintf(
-			"Could not retrieve K8s resource - '%s'", instanceDetails.Name))
+			"Could not retrieve K8s resource - '%s'", resName))
 		return err
 	}
 	if doesResExist {
 		log.Info(fmt.Sprintf(
-			"K8s resource - '%s' already exists", instanceDetails.Name))
+			"K8s resource - '%s' already exists", resName))
 		return nil
 	}
 
-	log.Info(fmt.Sprintf("K8s resource - '%s' does not exist. It will be recreated", instanceDetails.Name))
+	log.Info(fmt.Sprintf("K8s resource - '%s' does not exist. It will be recreated", resName))
 
 	k8sObj := r.identifyAndGetK8SObjToCreate(resName)
 	if k8sObj == nil {
 		log.Info(fmt.Sprintf("%s spec not set. Waiting for config_map controller to set it", resName))
 	}
 	if err = r.setControllerReference(r, k8sObj); err != nil {
 		log.Info(fmt.Sprintf(
```

Comment:
sorry, I can't resist pedantry. since we are logging an error that would cause a failure in the recreation processes, let's log this at error level?

Reply:
agree ;)